aws-datastore/src/main/java/com/amplifyframework/datastore/storage/sqlite/SQLiteStorageAdapter.java

```diff
@@ -475,10 +481,19 @@ private CreateSqlCommands getCreateCommands(@NonNull Set<Class<? extends Model>>
         }
         cursor.moveToFirst();
 
-        final ModelSchema modelSchema = ModelSchemaRegistry.singleton()
-                .getModelSchemaForModelClass(tableName);
-        final SQLiteTable sqliteTable = SQLiteTable.fromSchema(modelSchema);
-        final Map<String, SQLiteColumn> columns = sqliteTable.getColumns();
+        bindAllFieldsWithValues(object, preCompiledInsertStatement, cursor, columns);
+
+        if (!cursor.isClosed()) {
+            cursor.close();
+        }
+    }
+
+    private <T> void bindAllFieldsWithValues(@NonNull T object,
+                                             SQLiteStatement preCompiledInsertStatement,
+                                             Cursor cursor,
+                                             Map<String, SQLiteColumn> columns)
+            throws IllegalAccessException, DataStoreException {
+        final Iterator<Field> fieldIterator = FieldFinder.findFieldsIn(object.getClass()).iterator();
```

Comment:
`findFieldsIn` method relies on annotation, and I think this is the only place in `SQLiteStorageAdapter` that relies on annotation rather than model schema. This should be replaced with simple field iterator, since I already have logic to ignore irrelevant/unsupported fields inside the loop.

Reply:
Rebased with your changes.